apt/private/deb_resolve.bzl

```diff
@@ -64,20 +63,17 @@ def internal_resolve(rctx, yq_toolchain_prefix, manifest, include_transitive):
             constraint = version_constraint.parse_depends(dep_constraint).pop()
 
             rctx.report_progress("Resolving %s" % dep_constraint)
-            (package, dependencies, unmet_dependencies) = resolver.resolve_all(
-                name = constraint["name"],
-                version = constraint["version"],
-                arch = arch,
+            package, dependencies = resolver.resolve(
+                rctx,
+                arch,
+                constraint["name"],
+                constraint["version"],
                 include_transitive = include_transitive,
             )
 
             if not package:
                 fail("Unable to locate package `%s`" % dep_constraint)
 
-            if len(unmet_dependencies):
```

Comment:
I don't see how this a leak, that api is carefully designed this way to not depend much on rctx, now it's being passed as an argument to resolve. 

Previously its up to the caller to do what needs to be done with unmet dependencies but it now it's always warning!


And again if this needs to be change as a prefactor for a fix, please include this with that PR.

Reply:
I thought this codebase was the only user of itself, hence why I thought moving the warning to "inside" was equivalent to the current code but "not leaking". I agree on passing `rctx`, I also don't like it much but I thought it was better this way.

I'll revert this then.